v1.1/parallel-statement-execution.md

```diff
@@ -0,0 +1,125 @@
+
+## How Parallel Statement Execution Works
+
+SQL statements within a single transaction can be executed in parallel if the statements are independent. CockroachDB considers SQL statements within a single transaction to be independent if they operate on independent tables and their execution can be safely reordered without affecting their results. 
+
+Normal execution of SQL statements returns a successful return value immediately from the server. After the return value is received, the next SQL statement is executed. Appending the `RETURNING NOTHING` clause with SQL statements prevents the server from sending return values and allows CockroachDB to execute the statements in parallel. Thus to execute statements in parallel, append the `RETURNING NOTHING` suffix to the SQL statements. 
+
+### `RETURNING NOTHING` clause appended to dependent statements
+
+If two consecutive statements are not independent, and yet a `RETURNING NOTHING` clause is added to the statements, CockroachDB will detect the dependence and serialize the execution. You can thus use the `RETURNING NOTHING` clause with SQL statements without worrying about their dependence.
+
+### Delayed execution of sequentially-executed statements 
+
+The SQL statements to be executed in parallel should have a `RETURNING NOTHING` clause, while sequentially-executed statements do not have the `RETURNING NOTHING` clause. Consider a scenario where a sequentially-executed SQL statement follows a batch of parallely-executed statements within a single transaction. The sequentially-executed statement will be executed after all the parallel statements are done executing. Thus it may seem as if the sequentially-executed statement is taking longer to execute, but it's in fact waiting on the parallel statements. Even then, the total time required for parallel execution of statements followed by the sequential execution of the SQL statement should be less than time required for the sequential execution of all statements. 
+
+### Error message mismatch
+
+It is possible to see a mismatch in the error message being displayed and the statement being executed. Consider a scenario where a dependent statement is executed after a batch of independent statements, and one of the independent statements results in an error. Now because the `RETURNING NOTHING` clause is used with the independent statements, CockroachDB cannot return the error for the independent statement. The error message for the independent statement is then reported on the next dependent statement. Thus you might see a mismatch with the error message to the statement being executed. 
```

Comment:
Reworded:
It is possible to see a mismatch in the error message being displayed and the statement being executed. Consider a scenario where a sequentially-executed statement is executed after a batch of statements executed in parallel, and one of the parallel statements results in an error. Now because the `RETURNING NOTHING` clause is used with the parallel statements, CockroachDB cannot return the error for the parallel statement. The error message for the parallel statement is then reported on the next sequential statement. Thus you might see a mismatch with the error message to the statement being executed.